pkg/sql/executor.go

```diff
@@ -1126,10 +1126,12 @@ func (e *Executor) execStmtsInCurrentTxn(
 		queryMeta.ctx = txnState.Ctx
 		queryMeta.ctxCancel = txnState.cancel
 
+		// For parallel/async queries, we deregister queryMeta from these registries
+		// after execution finishes in the parallelizeQueue. SHOW TRANSACTION STATUSes
+		// are deregistered in their own code path. For all other queries,
+		// we deregister these in session.FinishPlan.
```

Comment:
Hi,
what is the point in leaving two places of query cancellation when it's pretty clear right now that it should happen after sending results?